src/tests/banners/bannerSelection.tsx

```diff
@@ -75,29 +141,29 @@ export const selectBannerTest = (
     targeting: BannerTargeting,
     pageTracking: BannerPageTracking,
     baseUrl: string,
-): Promise<BannerTestSelection | null> => {
-    return redeployedSinceLastClosed(targeting).then(hasRedeployed => {
-        if (hasRedeployed) {
-            const tests = [AusMomentContributionsBanner];
-            const testToRun = tests.find(
-                test =>
-                    targeting.alreadyVisitedCount >= test.minPageViews &&
-                    !targeting.shouldHideReaderRevenue &&
-                    !targeting.isPaidContent &&
-                    test.canRun(targeting, pageTracking),
-            );
-
-            if (testToRun) {
-                const variant = testToRun.variants[0]; // TODO - use mvt
+): BannerTestSelection | null => {
```

Comment:
I believe `selectBannerTest` will still need to return a `Promise<BannerTestSelection | null>` because of the async `redeployedSinceLastClosed` function call. 

I think as it is now this function will always return `null` as we're not returning the result of `tests.find( ... )`.

Also `tests.find` will always return the first item in the array, because `find` is not intended for asynchronous checks. It loops through the array until one of the elements results in a truthy value being returned. An object, including a `Promise` object, is truthy, and so the `find` stops on the first element in the `tests` array.

I think we'll need to do something different here. I've put together a gist which might help?

Find the first item in an array whose async test passes: https://gist.github.com/GHaberis/c20a0b3bd7aa8125dd115a54d79c72b8

Reply:
👍